PowerShell/JumpCloud Module/Public/Directory/Add-JCGsuiteMember.ps1

```diff
@@ -0,0 +1,106 @@
+function Add-JCGsuiteMember () {
+    param(
+        [Parameter( ValueFromPipelineByPropertyName, ParameterSetName = 'ByName', HelpMessage = 'The name of cloud directory instance')]
+        [String]$Name,
+        [Parameter( ValueFromPipelineByPropertyName, ParameterSetName = 'ByID', HelpMessage = 'The ID of cloud directory instance')]
+        [Alias('_id')]
+        [String]$ID,
+        [Parameter( ValueFromPipelineByPropertyName, HelpMessage = 'A username to add to the directory')]
+        [String]$Username,
+        [Parameter( ValueFromPipelineByPropertyName, HelpMessage = 'A UserID to add to the directory')]
+        [String]$UserID,
+        [Parameter( ValueFromPipelineByPropertyName, HelpMessage = 'A UserGroup ID to add to the directory')]
+        [String]$GroupID,
+        [Parameter( ValueFromPipelineByPropertyName, HelpMessage = 'A UserGroup name to add to the directory')]
+        [String]$GroupName
+    )
+    begin {
+        Write-Debug 'Verifying JCAPI Key'
+        if ($JCAPIKEY.length -ne 40) {
+            Connect-JCOnline
+        }
+        $resultsArray = [System.Collections.Generic.List[PSObject]]::new()
+        $DirectoryHash = Get-JcSdkDirectory | Where-Object type -EQ 'g_suite' | Select-Object id, name
+        if (($Username -or $UserID) -and ($GroupID -or $GroupName)) {
```

Comment:
I spent a bit of time thinking about this but I do think this is the only way to get around the many parameters we have here.

Reply:
In other words I saw this and thought we could do something else but I think this is fine :). Just wanted to call it out to generate conversation.